lib/realtime/rate_counter/rate_counter.ex

```diff
@@ -94,13 +94,12 @@ defmodule Realtime.RateCounter do
 
   @impl true
   def init(args) do
-    tenant_id = Keyword.fetch!(args, :tenant_id)
     id = Keyword.fetch!(args, :id)
     telem_opts = Keyword.get(args, :telemetry)
     every = Keyword.get(args, :tick, @tick)
     max_bucket_len = Keyword.get(args, :max_bucket_len, @max_bucket_len)
     idle_shutdown_ms = Keyword.get(args, :idle_shutdown, @idle_shutdown)
-
+    tenant_id = get_in(telem_opts, [:metadata, :tenant])
```

Comment:
This about this as if it's a library. We are passing metadata to telemetry options when we start up. We can just use the same metadata for Logger, and add the `project` and `external_id` when we init the RateCounter.